server/tests/src/test/java/org/infinispan/server/persistence/JdbcStringBasedCacheStorePassivationDisabled.java

```diff
@@ -0,0 +1,139 @@
+import org.infinispan.server.test.core.category.Persistence;
+import org.infinispan.server.test.core.persistence.Database;
+import org.infinispan.server.test.junit4.InfinispanServerRule;
+import org.infinispan.server.test.junit4.InfinispanServerTestMethodRule;
+import org.junit.ClassRule;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+
+/**
+ * Tests String-based jdbc cache store under the following circumstances:
+ * <p>
+ * passivation == false --all cache entries should always be also in the cache store
+ * preload == true --after server restart, all entries should appear in the cache immediately
+ * purge == false --all entries should remain in the cache store after server  * restart
+ * <p>
+ * Other attributes like singleton, shared, fetch-state do not make sense in single node cluster.
+ *
+ */
+@Category(Persistence.class)
+@RunWith(Parameterized.class)
+public class JdbcStringBasedCacheStorePassivationDisabled {
+
+    @ClassRule
+    public static InfinispanServerRule SERVERS = PersistenceIT.SERVERS;
+
+    @Rule
+    public InfinispanServerTestMethodRule SERVER_TEST = new InfinispanServerTestMethodRule(SERVERS);
+
+    private final Database database;
+
+    @Parameterized.Parameters(name = "{0}")
+    public static Collection<Object[]> data() {
+        String[] databaseTypes = PersistenceIT.DATABASE.getDatabaseTypes();
+        List<Object[]> params = new ArrayList<>(databaseTypes.length);
+        for (String databaseType : databaseTypes) {
+            params.add(new Object[]{databaseType});
+        }
+        return params;
+    }
+
+    public JdbcStringBasedCacheStorePassivationDisabled(String databaseType) {
+        this.database = PersistenceIT.DATABASE.getDatabase(databaseType);
+    }
+
+    @Test(timeout = 600000)
+    public void testFailoverWithPassivationDisabled() throws Exception {
+        JdbcConfigurationUtil jdbcUtil = new JdbcConfigurationUtil(CacheMode.REPL_SYNC)
+                .createPersistenceConfiguration(database, false, true)
+                .setLockingConfigurations();
+        RemoteCache<String, String> cache = SERVER_TEST.hotrod().withServerConfiguration(jdbcUtil.getConfigurationBuilder()).create();
+        TableManipulation table = new TableManipulation(cache.getName(), jdbcUtil.getPersistenceConfiguration(), jdbcUtil.getConfigurationBuilder());
+
+        cache.put("k1", "v1");
+        cache.put("k2", "v2");
+        // test passivation==false, database should contain all entries which are in the cache
+        assertNotNull(table.getValueByKey("k1"));
+        assertNotNull(table.getValueByKey("k2"));
+
+        SERVERS.getServerDriver().stop(0);
+        SERVERS.getServerDriver().restart(0);
+
+        assertNotNull(table.getValueByKey("k1"));
+        assertNotNull(table.getValueByKey("k2"));
+        assertEquals("v1", cache.get("k1"));
```

Comment:
It's probably worth ensuring that the key is not in-memory first, so `assertNull(cache.withFlags(Flag.SKIP_CACHE_LOAD).get("k1")` followed by `assertEquals("v1", cache.get("k1"));`